task1Daniv/stringlist.cpp

```diff
@@ -0,0 +1,250 @@
+#include "stringlist.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+const int dataIndex = 0;
+const int nextNodeIndex = 1;
+
+List initialization()
+{
+    List temp = static_cast<char**>(malloc(sizeof(List) * 2));
+
+    if(temp != nullptr)
+    {
+        temp[dataIndex] = nullptr;
+        temp[nextNodeIndex] = nullptr;
+    }
+    return  temp;
+}
+
+void push(List &list, const String &string)
```

Comment:
I recommend to pass list by value instead of passing it by reference because this function works with pointers and this shall eliminate possible changes.